pkg/operator/ceph/csi/spec.go

```diff
@@ -394,11 +405,99 @@ func StartCSIDrivers(namespace string, clientset kubernetes.Interface, ver *vers
 	return nil
 }
 
-func StopCSIDrivers(namespace string, clientset kubernetes.Interface) error {
-	logger.Warningf("Skipped removing the CSI driver")
+func StopCSIDrivers(namespace string, clientset kubernetes.Interface, ver *version.Info) error {
+
+	deployProvSTS := ver.Major > KubeMinMajor || (ver.Major == KubeMinMajor && ver.Minor < provDeploymentSuppVersion)
```

Comment:
In the interest of keeping this cleanup method simple, what about the following?
- Treat this as a best-effort cleanup. Let's attempt to delete all of the resources instead of returning early if there is an error.
- In the log it seems sufficient to have one message that we're removing the csi drivers. No other info or warnings seem necessary if anything fails. If there are failures, they will see the dangling resources anyway.
- No need to check any k8s versions or flags for whether the drivers were enabled. If we ignore any errors, there is no side effect to just attempting to delete them all.
It may seem counter-intuitive to ignore errors or not log anything, but this is the same behavior as if the ownerReferences would have worked in this scenario.